[APM] Correlations Beta (#86477)

ogupte opened the pull request:
Closes #86477.
Closes #91080.
Closes #90766.
Closes #90622.

![Screen Shot 2021-02-12 at 3 15 19 AM](https://user-images.githubusercontent.com/1967266/107751487-503c6780-6cd2-11eb-850c-072c0845cab2.png)
![Screen Shot 2021-02-16 at 4 00 57 PM](https://user-images.githubusercontent.com/1967266/108121078-144d2d80-7057-11eb-8e72-58b83a105b03.png)
![Screen Shot 2021-02-16 at 3 58 49 PM](https://user-images.githubusercontent.com/1967266/108121080-14e5c400-7057-11eb-8f96-c34a1f07bc3a.png)

ogupte committed:
[APM] Correlations GA (#86477)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102832/)
* Commit: 4892a72eb0e4cc72a0b80fc2c08b77de5091cd70
* This comment will update when the build is complete

### Failed CI Steps
* [Build X-Pack Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102832/execution/node/265/log)
* [Report failed tests, if necessary](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102832/execution/node/277/log)
* [X-Pack Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102832/execution/node/246/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102832/","number":"102832","commit":"4892a72eb0e4cc72a0b80fc2c08b77de5091cd70"}]}
PIPELINE-->

ogupte committed:
polish and improvements to correlations UI

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104068/)
* Commit: dc83a124f1df240053127484352d50f694352868
* This comment will update when the build is complete

### Failed CI Steps
* [Build X-Pack Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104068/execution/node/257/log)
* [Report failed tests, if necessary](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104068/execution/node/277/log)
* [X-Pack Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104068/execution/node/270/log)

### History
* :broken_heart: [Build #102832](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102832/) failed 4892a72eb0e4cc72a0b80fc2c08b77de5091cd70

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102832/","number":"102832","commit":"4892a72eb0e4cc72a0b80fc2c08b77de5091cd70"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104068/","number":"104068","commit":"dc83a124f1df240053127484352d50f694352868"}]}
PIPELINE-->

ogupte committed:
more improvements and polish

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104427/)
* Commit: 82d477e6ea9c5ddad692bdf88ac85cbc6c99a6e1
* This comment will update when the build is complete

### Failed CI Steps
* [Build X-Pack Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104427/execution/node/257/log)
* [Report failed tests, if necessary](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104427/execution/node/277/log)
* [X-Pack Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104427/execution/node/238/log)

### History
* :broken_heart: [Build #104068](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104068/) failed dc83a124f1df240053127484352d50f694352868
* :broken_heart: [Build #102832](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102832/) failed 4892a72eb0e4cc72a0b80fc2c08b77de5091cd70

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102832/","number":"102832","commit":"4892a72eb0e4cc72a0b80fc2c08b77de5091cd70"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104068/","number":"104068","commit":"dc83a124f1df240053127484352d50f694352868"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104427/","number":"104427","commit":"82d477e6ea9c5ddad692bdf88ac85cbc6c99a6e1"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104442/)
* Commit: 63da7e61112b2c4dbc72b62f3f74a64454ce789d
* This comment will update when the build is complete

### Failed CI Steps
* [Build X-Pack Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104442/execution/node/257/log)
* [Report failed tests, if necessary](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104442/execution/node/277/log)
* [X-Pack Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104442/execution/node/270/log)

### History
* :broken_heart: [Build #104427](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104427/) failed 82d477e6ea9c5ddad692bdf88ac85cbc6c99a6e1
* :broken_heart: [Build #104068](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104068/) failed dc83a124f1df240053127484352d50f694352868
* :broken_heart: [Build #102832](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102832/) failed 4892a72eb0e4cc72a0b80fc2c08b77de5091cd70

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102832/","number":"102832","commit":"4892a72eb0e4cc72a0b80fc2c08b77de5091cd70"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104068/","number":"104068","commit":"dc83a124f1df240053127484352d50f694352868"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104427/","number":"104427","commit":"82d477e6ea9c5ddad692bdf88ac85cbc6c99a6e1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104442/","number":"104442","commit":"63da7e61112b2c4dbc72b62f3f74a64454ce789d"}]}
PIPELINE-->

sqren commented on x-pack/plugins/apm/public/components/app/Correlations/correlations_table.tsx:
```diff
@@ -29,22 +33,27 @@ interface Props<T> {
+      ),
       render: (_: any, term: T) => {
-        return <EuiCode>{Math.round(term.score)}</EuiCode>;
+        return <EuiBadge>{Math.round(term.score)}</EuiBadge>;
```
I think we talked about showing the score visually - similar to how impact is displayed on the transactions table. I think impact should be calculated as:
```
score / maxScoreInCurrentSet
```

sqren replied:
Actually, another approach I think is better (it seems arbitrary but I think it'll yield better results):
```
const max = Math.max(maxScore, medianScore * 2);
const impact = score / max;
```

Try it out an see what works best.

elasticmachine commented:
Pinging @elastic/apm-ui (Team:apm)

ogupte committed:
added impact bar

ogupte committed:
added descriptions

ogupte committed:
make custom field persistence be unique per service

ogupte committed:
make custom threshold unique per service in latency correlations

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105252/)
* Commit: c535b4a2de7bee3daa5a522e40e50df0bf24aa00
* This comment will update when the build is complete

### Failed CI Steps
* [Build X-Pack Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105252/execution/node/265/log)
* [Report failed tests, if necessary](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105252/execution/node/277/log)
* [X-Pack Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105252/execution/node/246/log)

### History
* :broken_heart: [Build #104442](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104442/) failed 63da7e61112b2c4dbc72b62f3f74a64454ce789d
* :broken_heart: [Build #104427](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104427/) failed 82d477e6ea9c5ddad692bdf88ac85cbc6c99a6e1
* :broken_heart: [Build #104068](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104068/) failed dc83a124f1df240053127484352d50f694352868
* :broken_heart: [Build #102832](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102832/) failed 4892a72eb0e4cc72a0b80fc2c08b77de5091cd70

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102832/","number":"102832","commit":"4892a72eb0e4cc72a0b80fc2c08b77de5091cd70"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104068/","number":"104068","commit":"dc83a124f1df240053127484352d50f694352868"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104427/","number":"104427","commit":"82d477e6ea9c5ddad692bdf88ac85cbc6c99a6e1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104442/","number":"104442","commit":"63da7e61112b2c4dbc72b62f3f74a64454ce789d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105252/","number":"105252","commit":"c535b4a2de7bee3daa5a522e40e50df0bf24aa00"}]}
PIPELINE-->

sqren commented on x-pack/plugins/apm/public/components/app/Correlations/useFieldNames.ts:
```diff
@@ -0,0 +1,75 @@
+) {
+  return isRumAgent
+    ? [
+        ...take(getMatchingFieldNames(indexPattern, 'labels.'), 6),
```
Why only the first 6? And do you mind using vanilla slice? I at least didn't remember what `take` did. I'd expect most js developers to understand slice.

```suggestion
        ...getMatchingFieldNames(indexPattern, 'labels.').slice(0, 6)
```

sqren replied:
Btw. this can be extracted to
```
const labelFields = getMatchingFieldNames(indexPattern, 'labels.').slice(0, 6);
```

And then re-used

sqren commented on x-pack/plugins/apm/public/components/app/Correlations/useFieldNames.ts:
```diff
@@ -0,0 +1,75 @@
+  if (!indexPattern) {
+    return [];
+  }
+  return indexPattern.fields.reduce(
+    (acc, { name }) => (name.startsWith(inputValue) ? [...acc, name] : acc),
+    [] as string[]
+  );
```
map + filter is more readable.

```suggestion
  return indexPattern.fields
    .filter(({name}) => name.startsWith(inputValue))
    .map(({name}) => name)
```

I don't think the perf difference is significant but haven't checked.

sqren commented on x-pack/plugins/apm/public/components/app/Correlations/useFieldNames.ts:
```diff
@@ -0,0 +1,75 @@
+    ProcessorEvent.transaction
+  );
+
+  const [defaultFieldNames, setDefaultFieldNames] = useState(
+    getDefaultFieldNames(indexPattern, isRumAgent)
+  );
```
It feels off that this is stateful. If for perf reasons you can use `useMemo`. 
But first try to strip it down and see if it's needed

ogupte replied:
It's stateful because `useDynamicIndexPatternFetcher` is an async fetcher, so it needs to trigger a re-render when the response comes back.

sqren commented on x-pack/plugins/apm/public/components/app/Correlations/custom_fields.tsx:
```diff
@@ -0,0 +1,155 @@
+}: Props) {
+  const { defaultFieldNames, getSuggestions } = useFieldNames();
+  const [suggestedFieldNames, setSuggestedFieldNames] = useState(
+    getSuggestions('')
```
This looks odd. Why calling it with an empty string?

sqren commented on x-pack/plugins/apm/public/components/app/Correlations/useFieldNames.ts:
```diff
@@ -0,0 +1,75 @@
+
+  const getSuggestions = useMemo(
+    () =>
+      memoize((searchValue: string) =>
```
Is it neccessary to use both _.memoize and useMemo ?

ogupte committed:
adds telemetry for apm correlations feature. Events:
 - 'show_correlations_flyout'
 - 'customize_correlations_fields'
 - 'select_significant_term'

ogupte committed:
adds more telemetry for correlations (#90622)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105397/)
* Commit: c52deb3df15c1b72a059c796b8496054b52eaa59
* This comment will update when the build is complete

### Failed CI Steps
* [Build X-Pack Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105397/execution/node/257/log)
* [Report failed tests, if necessary](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105397/execution/node/277/log)
* [X-Pack Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105397/execution/node/238/log)

### History
* :broken_heart: [Build #105252](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105252/) failed c535b4a2de7bee3daa5a522e40e50df0bf24aa00
* :broken_heart: [Build #104442](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104442/) failed 63da7e61112b2c4dbc72b62f3f74a64454ce789d
* :broken_heart: [Build #104427](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104427/) failed 82d477e6ea9c5ddad692bdf88ac85cbc6c99a6e1
* :broken_heart: [Build #104068](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104068/) failed dc83a124f1df240053127484352d50f694352868
* :broken_heart: [Build #102832](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102832/) failed 4892a72eb0e4cc72a0b80fc2c08b77de5091cd70

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102832/","number":"102832","commit":"4892a72eb0e4cc72a0b80fc2c08b77de5091cd70"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104068/","number":"104068","commit":"dc83a124f1df240053127484352d50f694352868"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104427/","number":"104427","commit":"82d477e6ea9c5ddad692bdf88ac85cbc6c99a6e1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104442/","number":"104442","commit":"63da7e61112b2c4dbc72b62f3f74a64454ce789d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105252/","number":"105252","commit":"c535b4a2de7bee3daa5a522e40e50df0bf24aa00"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105397/","number":"105397","commit":"c52deb3df15c1b72a059c796b8496054b52eaa59"}]}
PIPELINE-->

sqren commented on x-pack/plugins/apm/public/components/shared/search_bar.tsx:
```diff
@@ -22,15 +27,26 @@ const SearchBarFlexGroup = styled(EuiFlexGroup)`
+  const { uiSettings } = useApmPluginContext().core;
+  const license = useLicenseContext();
+  const hasCorrelationsAccess =
+    uiSettings.get(enableCorrelations) && isActivePlatinumLicense(license);
```
Are we going to remove this setting? What advantages are there to leave it in?

ogupte replied:
removed

sqren commented on x-pack/plugins/apm/server/lib/correlations/get_correlations_for_failed_transactions/index.ts:
```diff
@@ -121,7 +121,7 @@ export async function getErrorRateTimeSeries({
 }) {
   const { start, end, apmEventClient } = setup;
-  const { intervalString } = getBucketSize({ start, end, numBuckets: 30 });
+  const { intervalString } = getBucketSize({ start, end, numBuckets: 15 });
```
I think this will still return a dynamic number of buckets to aim for a "nice" bucket size

ogupte replied:
We'll fix this in a separate issue: https://github.com/elastic/kibana/issues/91574

sqren commented on x-pack/plugins/apm/server/lib/correlations/get_correlations_for_slow_transactions/get_latency_distribution.ts:
```diff
@@ -107,7 +107,14 @@ export async function getLatencyDistribution({
-    return distribution.dist_filtered_by_latency.buckets.map((bucket) => ({
+
+    // remove trailing buckets that are empty and out of bounds of the desired number of buckets
+    const buckets = dropRightWhile(
+      distribution.dist_filtered_by_latency.buckets,
+      (bucket, index) => bucket.doc_count === 0 && index > intervalBuckets - 1
+    );
```
I don't understand why we are doing this. If we want fewer bucket returned we shouldn't we specify a larger interval?

sqren replied:
Btw. all this should matter less if we use a logarithmic scale

ogupte replied:
I created a separate issue to address it https://github.com/elastic/kibana/issues/91572

ogupte replied:
I created a separate issue for the distribution scale: https://github.com/elastic/kibana/issues/91574

ogupte committed:
removes the raw score column

ogupte committed:
replaces experiemental callout with beta badge

ogupte committed:
replaces threshold number input with percentile option selector

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106184/)
* Commit: 1c99c5c0c3a5ba8f19633bd7a01698ac53838f57
* This comment will update when the build is complete

### Failed CI Steps
* [Build X-Pack Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106184/execution/node/265/log)
* [Report failed tests, if necessary](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106184/execution/node/277/log)
* [X-Pack Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106184/execution/node/246/log)

### History
* :broken_heart: [Build #105397](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105397/) failed c52deb3df15c1b72a059c796b8496054b52eaa59
* :broken_heart: [Build #105252](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105252/) failed c535b4a2de7bee3daa5a522e40e50df0bf24aa00
* :broken_heart: [Build #104442](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104442/) failed 63da7e61112b2c4dbc72b62f3f74a64454ce789d
* :broken_heart: [Build #104427](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104427/) failed 82d477e6ea9c5ddad692bdf88ac85cbc6c99a6e1
* :broken_heart: [Build #104068](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104068/) failed dc83a124f1df240053127484352d50f694352868

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104068/","number":"104068","commit":"dc83a124f1df240053127484352d50f694352868"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104427/","number":"104427","commit":"82d477e6ea9c5ddad692bdf88ac85cbc6c99a6e1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104442/","number":"104442","commit":"63da7e61112b2c4dbc72b62f3f74a64454ce789d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105252/","number":"105252","commit":"c535b4a2de7bee3daa5a522e40e50df0bf24aa00"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105397/","number":"105397","commit":"c52deb3df15c1b72a059c796b8496054b52eaa59"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106184/","number":"106184","commit":"1c99c5c0c3a5ba8f19633bd7a01698ac53838f57"}]}
PIPELINE-->

ogupte committed:
improvements to latency correlations scoring and percentage reporting

ogupte committed:
Merge branch 'master' into apm-86477-correlations-ga

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106353/)
* Commit: f8783011fd72394d5b5f34081380f2f0dbfa10ad
* This comment will update when the build is complete

### Failed CI Steps
* [Build X-Pack Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106353/execution/node/182/log)
* [Report failed tests, if necessary](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106353/execution/node/202/log)
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106353/execution/node/194/log)

### History
* :broken_heart: [Build #106184](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106184/) failed 1c99c5c0c3a5ba8f19633bd7a01698ac53838f57
* :broken_heart: [Build #105397](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105397/) failed c52deb3df15c1b72a059c796b8496054b52eaa59
* :broken_heart: [Build #105252](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105252/) failed c535b4a2de7bee3daa5a522e40e50df0bf24aa00
* :broken_heart: [Build #104442](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104442/) failed 63da7e61112b2c4dbc72b62f3f74a64454ce789d
* :broken_heart: [Build #104427](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104427/) failed 82d477e6ea9c5ddad692bdf88ac85cbc6c99a6e1

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104427/","number":"104427","commit":"82d477e6ea9c5ddad692bdf88ac85cbc6c99a6e1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104442/","number":"104442","commit":"63da7e61112b2c4dbc72b62f3f74a64454ce789d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105252/","number":"105252","commit":"c535b4a2de7bee3daa5a522e40e50df0bf24aa00"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105397/","number":"105397","commit":"c52deb3df15c1b72a059c796b8496054b52eaa59"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106184/","number":"106184","commit":"1c99c5c0c3a5ba8f19633bd7a01698ac53838f57"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106353/","number":"106353","commit":"f8783011fd72394d5b5f34081380f2f0dbfa10ad"}]}
PIPELINE-->

ogupte committed:
Merge branch 'master' into apm-86477-correlations-ga

ogupte committed:
removes the 'apm:enableCorrelations' UI setting

ogupte committed:
- rename useFieldNames.ts -> use_field_names.ts
- filter out fields that are not type 'keyword'
- feedback improvements

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106928/)
* Commit: 429bc7b00fb1833b1acccb9d57762c661d320e71
* This comment will update when the build is complete

### Failed CI Steps
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106928/execution/node/109/log)
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106928/execution/node/169/log)

### History
* :broken_heart: [Build #106353](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106353/) failed f8783011fd72394d5b5f34081380f2f0dbfa10ad
* :broken_heart: [Build #106184](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106184/) failed 1c99c5c0c3a5ba8f19633bd7a01698ac53838f57
* :broken_heart: [Build #105397](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105397/) failed c52deb3df15c1b72a059c796b8496054b52eaa59
* :broken_heart: [Build #105252](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105252/) failed c535b4a2de7bee3daa5a522e40e50df0bf24aa00
* :broken_heart: [Build #104442](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104442/) failed 63da7e61112b2c4dbc72b62f3f74a64454ce789d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104442/","number":"104442","commit":"63da7e61112b2c4dbc72b62f3f74a64454ce789d"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105252/","number":"105252","commit":"c535b4a2de7bee3daa5a522e40e50df0bf24aa00"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105397/","number":"105397","commit":"c52deb3df15c1b72a059c796b8496054b52eaa59"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106184/","number":"106184","commit":"1c99c5c0c3a5ba8f19633bd7a01698ac53838f57"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106353/","number":"106353","commit":"f8783011fd72394d5b5f34081380f2f0dbfa10ad"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106928/","number":"106928","commit":"429bc7b00fb1833b1acccb9d57762c661d320e71"}]}
PIPELINE-->

ogupte committed:
Fixes casing issue for the 'correlations' dir

ogupte committed:
[APM] Moves correlations button to service details tabslist row (#91080)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106968/)
* Commit: cab10a2143659c00b210038c2b7da8ed1a21c923
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup1](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106968/execution/node/681/log)
* [Execute xpack-kibana-ciGroup1](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106968/execution/node/876/log)

### History
* :broken_heart: [Build #106928](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106928/) failed 429bc7b00fb1833b1acccb9d57762c661d320e71
* :broken_heart: [Build #106353](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106353/) failed f8783011fd72394d5b5f34081380f2f0dbfa10ad
* :broken_heart: [Build #106184](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106184/) failed 1c99c5c0c3a5ba8f19633bd7a01698ac53838f57
* :broken_heart: [Build #105397](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105397/) failed c52deb3df15c1b72a059c796b8496054b52eaa59
* :broken_heart: [Build #105252](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105252/) failed c535b4a2de7bee3daa5a522e40e50df0bf24aa00

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105252/","number":"105252","commit":"c535b4a2de7bee3daa5a522e40e50df0bf24aa00"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105397/","number":"105397","commit":"c52deb3df15c1b72a059c796b8496054b52eaa59"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106184/","number":"106184","commit":"1c99c5c0c3a5ba8f19633bd7a01698ac53838f57"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106353/","number":"106353","commit":"f8783011fd72394d5b5f34081380f2f0dbfa10ad"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106928/","number":"106928","commit":"429bc7b00fb1833b1acccb9d57762c661d320e71"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106968/","number":"106968","commit":"cab10a2143659c00b210038c2b7da8ed1a21c923"}]}
PIPELINE-->

ogupte committed:
[APM] Adds license check for correlations (#90766)

ogupte committed:
[APM] Adds metrics tracking for correlations views and license prompts (#90622)

ogupte commented:
jenkins run the e2e

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107005/)
* Commit: 0f09c0bdbf1619f189584c1b04fe62ea3254a9c9
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup1](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107005/execution/node/681/log)
* [Execute xpack-kibana-ciGroup1](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107005/execution/node/876/log)

### History
* :broken_heart: [Build #106968](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106968/) failed cab10a2143659c00b210038c2b7da8ed1a21c923
* :broken_heart: [Build #106928](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106928/) failed 429bc7b00fb1833b1acccb9d57762c661d320e71
* :broken_heart: [Build #106353](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106353/) failed f8783011fd72394d5b5f34081380f2f0dbfa10ad
* :broken_heart: [Build #106184](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106184/) failed 1c99c5c0c3a5ba8f19633bd7a01698ac53838f57
* :broken_heart: [Build #105397](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105397/) failed c52deb3df15c1b72a059c796b8496054b52eaa59

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105397/","number":"105397","commit":"c52deb3df15c1b72a059c796b8496054b52eaa59"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106184/","number":"106184","commit":"1c99c5c0c3a5ba8f19633bd7a01698ac53838f57"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106353/","number":"106353","commit":"f8783011fd72394d5b5f34081380f2f0dbfa10ad"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106928/","number":"106928","commit":"429bc7b00fb1833b1acccb9d57762c661d320e71"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106968/","number":"106968","commit":"cab10a2143659c00b210038c2b7da8ed1a21c923"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107005/","number":"107005","commit":"0f09c0bdbf1619f189584c1b04fe62ea3254a9c9"}]}
PIPELINE-->

TinaHeiligers commented on src/plugins/telemetry/schema/oss_plugins.json:
```diff
@@ -4384,9 +4384,6 @@
         "apm:enableServiceOverview": {
           "type": "boolean"
-        },
-        "apm:enableCorrelations": {
```
FYI, You'd need to open a corresponding mapping update request in the [infra repo](https://github.com/elastic/infra/issues/new/choose) to have this key removed from the telemetry mappings.
If you don't care that it's not there in versions from 7.12, then you won't need to open an issue for that.

ogupte replied:
We'll probably remove it from the mapping in the infra repo in 7.13. That way we'll be able to tell if users are still using it as a custom UI Setting for version 7.12.

TinaHeiligers reviewed (approved):
I reviewed both the `ui_settings` key removal and the usage collection schema update for telemetry.
LGTM

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107017/)
* Commit: b5f69ab0d94342d548a63d48f87ab5195b229cf5
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup1](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107017/execution/node/681/log)
* [Execute xpack-kibana-ciGroup1](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107017/execution/node/876/log)

### History
* :broken_heart: [Build #107005](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107005/) failed 0f09c0bdbf1619f189584c1b04fe62ea3254a9c9
* :broken_heart: [Build #106968](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106968/) failed cab10a2143659c00b210038c2b7da8ed1a21c923
* :broken_heart: [Build #106928](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106928/) failed 429bc7b00fb1833b1acccb9d57762c661d320e71
* :broken_heart: [Build #106353](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106353/) failed f8783011fd72394d5b5f34081380f2f0dbfa10ad
* :broken_heart: [Build #106184](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106184/) failed 1c99c5c0c3a5ba8f19633bd7a01698ac53838f57

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106184/","number":"106184","commit":"1c99c5c0c3a5ba8f19633bd7a01698ac53838f57"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106353/","number":"106353","commit":"f8783011fd72394d5b5f34081380f2f0dbfa10ad"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106928/","number":"106928","commit":"429bc7b00fb1833b1acccb9d57762c661d320e71"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106968/","number":"106968","commit":"cab10a2143659c00b210038c2b7da8ed1a21c923"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107005/","number":"107005","commit":"0f09c0bdbf1619f189584c1b04fe62ea3254a9c9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107017/","number":"107017","commit":"b5f69ab0d94342d548a63d48f87ab5195b229cf5"}]}
PIPELINE-->

ogupte committed:
Updated the API integration tests to check for new default fields and 15 buckets

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107029/)
* Commit: cd3407ed98ba1b01160ce87311169681149cb2a8
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup1](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107029/execution/node/681/log)
* [Execute xpack-kibana-ciGroup1](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107029/execution/node/876/log)

### History
* :broken_heart: [Build #107017](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107017/) failed b5f69ab0d94342d548a63d48f87ab5195b229cf5
* :broken_heart: [Build #107005](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107005/) failed 0f09c0bdbf1619f189584c1b04fe62ea3254a9c9
* :broken_heart: [Build #106968](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106968/) failed cab10a2143659c00b210038c2b7da8ed1a21c923
* :broken_heart: [Build #106928](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106928/) failed 429bc7b00fb1833b1acccb9d57762c661d320e71
* :broken_heart: [Build #106353](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106353/) failed f8783011fd72394d5b5f34081380f2f0dbfa10ad

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106353/","number":"106353","commit":"f8783011fd72394d5b5f34081380f2f0dbfa10ad"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106928/","number":"106928","commit":"429bc7b00fb1833b1acccb9d57762c661d320e71"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106968/","number":"106968","commit":"cab10a2143659c00b210038c2b7da8ed1a21c923"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107005/","number":"107005","commit":"0f09c0bdbf1619f189584c1b04fe62ea3254a9c9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107017/","number":"107017","commit":"b5f69ab0d94342d548a63d48f87ab5195b229cf5"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107029/","number":"107029","commit":"cd3407ed98ba1b01160ce87311169681149cb2a8"}]}
PIPELINE-->

ogupte commented:
jenkins run the e2e

sqren commented on x-pack/plugins/apm/server/lib/correlations/get_correlations_for_slow_transactions/index.ts:
```diff
@@ -70,14 +72,21 @@ export async function getCorrelationsForSlowTransactions({
-                {
-                  range: {
-                    [TRANSACTION_DURATION]: { gte: durationForPercentile },
+              filter: backgroundFilters,
```
What's the reason for removing the range query for duration? Will removing it not hurt perf since more documents are retrieved?

ogupte replied:
moving the range query to the `function_score` provided us with a score that we needed to order foreground results that were most important to us (those with higher durations)

sqren commented on x-pack/plugins/apm/server/lib/correlations/get_correlations_for_slow_transactions/index.ts:
```diff
@@ -87,7 +96,20 @@ export async function getCorrelationsForSlowTransactions({
+                        ...backgroundFilters,
+                        {
+                          range: {
+                            [TRANSACTION_DURATION]: {
+                              lt: durationForPercentile,
+                            },
```
Isn't the foreground supposed to be a subset of the background?

sqren replied:
... if not you may have to set this 
```ts
"background_is_superset": false
```

ogupte replied:
We found that the existing background filter included too much uninteresting data and it was skewing the results of significant terms agg to be not as helpful see(https://youtu.be/azP15yvbOBA)

sqren commented on x-pack/plugins/apm/server/lib/correlations/process_significant_term_aggs.ts:
```diff
@@ -12,43 +12,65 @@ import {
+    medianIndex < halfSize
+      ? sortedScores[medianIndex]
+      : (sortedScores[medianIndex - 1] + sortedScores[medianIndex]) / 2;
+
+  return Math.max(maxScore, medianScore * 2);
+}
+
 export function processSignificantTermAggs({
   sigTermAggs,
-  thresholdPercentage,
 }: {
   sigTermAggs: Record<string, SigTermAgg>;
-  thresholdPercentage: number;
 }) {
   const significantTerms = Object.entries(sigTermAggs).flatMap(
     ([fieldName, agg]) => {
       return agg.buckets.map((bucket) => ({
         fieldName,
         fieldValue: bucket.key,
-        bgCount: bucket.bg_count,
-        fgCount: bucket.doc_count,
+        fieldCount: agg.doc_count,
+        valueCount: bucket.doc_count,
```
Not sure about this change. Can you elaborate a bit?

ogupte replied:
the previous `bgCount` wasn't needed since it represents only values for that field/value pair, when we really wanted to express the percentage in terms of the total number of background documents the aggregation applies to. E.g. `user_agent.name: HeadlessChrome` accounts for only 8% of all documents that meet the 75% threshold.

ogupte replied:
previously it showed something like "80% of documents with `user_agent.name: HeadlessChrome` exist in the 75th percentile threshold", so it's a question of what is easier to understand for the user.

ogupte commented:
jenkins run the e2e

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107059/)
* Commit: 7eae6b950898b78a306fc81c52cfd7cb25577946

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/58849bca385da66ed9e9c3a2745c52037b14b15b) | [after](https://github.com/elastic/kibana/commit/7eae6b950898b78a306fc81c52cfd7cb25577946) | diff |
| --- | --- | --- | --- |
| `apm` | 1731 | 1733 | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/58849bca385da66ed9e9c3a2745c52037b14b15b) | [after](https://github.com/elastic/kibana/commit/7eae6b950898b78a306fc81c52cfd7cb25577946) | diff |
| --- | --- | --- | --- |
| `apm` | 5.2MB | 5.2MB | +9.9KB |

### History
* :broken_heart: [Build #107029](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107029/) failed cd3407ed98ba1b01160ce87311169681149cb2a8
* :broken_heart: [Build #107017](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107017/) failed b5f69ab0d94342d548a63d48f87ab5195b229cf5
* :broken_heart: [Build #107005](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107005/) failed 0f09c0bdbf1619f189584c1b04fe62ea3254a9c9
* :broken_heart: [Build #106968](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106968/) failed cab10a2143659c00b210038c2b7da8ed1a21c923
* :broken_heart: [Build #106928](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106928/) failed 429bc7b00fb1833b1acccb9d57762c661d320e71

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106928/","number":"106928","commit":"429bc7b00fb1833b1acccb9d57762c661d320e71"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106968/","number":"106968","commit":"cab10a2143659c00b210038c2b7da8ed1a21c923"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107005/","number":"107005","commit":"0f09c0bdbf1619f189584c1b04fe62ea3254a9c9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107017/","number":"107017","commit":"b5f69ab0d94342d548a63d48f87ab5195b229cf5"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107029/","number":"107029","commit":"cd3407ed98ba1b01160ce87311169681149cb2a8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107059/","number":"107059","commit":"7eae6b950898b78a306fc81c52cfd7cb25577946"}]}
PIPELINE-->

ogupte commented:
jenkins run the e2e

ogupte commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'master' into apm-86477-correlations-ga

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107099/)
* Commit: 50cd63a72ed98d04214697eb7ef66c7ef089d01e

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/63ac0f74be6c0339d8ec63a6679ee3ebb93e153a) | [after](https://github.com/elastic/kibana/commit/50cd63a72ed98d04214697eb7ef66c7ef089d01e) | diff |
| --- | --- | --- | --- |
| `apm` | 1731 | 1733 | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/63ac0f74be6c0339d8ec63a6679ee3ebb93e153a) | [after](https://github.com/elastic/kibana/commit/50cd63a72ed98d04214697eb7ef66c7ef089d01e) | diff |
| --- | --- | --- | --- |
| `apm` | 5.2MB | 5.2MB | +9.9KB |

### History
* :green_heart: [Build #107059](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107059/) succeeded 7eae6b950898b78a306fc81c52cfd7cb25577946
* :broken_heart: [Build #107029](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107029/) failed cd3407ed98ba1b01160ce87311169681149cb2a8
* :broken_heart: [Build #107017](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107017/) failed b5f69ab0d94342d548a63d48f87ab5195b229cf5
* :broken_heart: [Build #107005](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107005/) failed 0f09c0bdbf1619f189584c1b04fe62ea3254a9c9
* :broken_heart: [Build #106968](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106968/) failed cab10a2143659c00b210038c2b7da8ed1a21c923

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106968/","number":"106968","commit":"cab10a2143659c00b210038c2b7da8ed1a21c923"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107005/","number":"107005","commit":"0f09c0bdbf1619f189584c1b04fe62ea3254a9c9"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107017/","number":"107017","commit":"b5f69ab0d94342d548a63d48f87ab5195b229cf5"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107029/","number":"107029","commit":"cd3407ed98ba1b01160ce87311169681149cb2a8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107059/","number":"107059","commit":"7eae6b950898b78a306fc81c52cfd7cb25577946"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/107099/","number":"107099","commit":"50cd63a72ed98d04214697eb7ef66c7ef089d01e"}]}
PIPELINE-->
